docs/src/pages/components/masonry/ImageMasonry.tsx

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as React from 'react';
+import Masonry from '@material-ui/lab/Masonry';
+import MasonryItem from '@material-ui/lab/MasonryItem';
+
+export default function ImageMasonry(): JSX.Element {
+  return (
+    <Masonry
+      cols={{ xs: 3, sm: 5, md: 7 }}
+      spacing={{ xs: 1, sm: 2, md: 3 }}
+      sx={{ height: 300 }}
+    >
+      {imgData.map((item, idx) => (
+        <MasonryItem key={idx} height={item.height}>
+          <img alt={item.title} src={item.img} />
+        </MasonryItem>
+      ))}
+    </Masonry>
+  );
+}
+
+const imgData = [
+  {
+    img: 'https://images.unsplash.com/photo-1529655683826-aba9b3e77383?ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8bG9uZG9ufGVufDB8fDB8fA%3D%3D&ixlib=rb-1.2.1&w=1000&q=80',
```

Comment:
For the image data, we should be able to improve it:

1. Could we share the same data set as with https://deploy-preview-27169--material-ui.netlify.app/components/image-list/#standard-image-list? So we don't have to maintain two different sets.
2. Also, could you replicate the image loading strategy? (lazy loading, srcset retina adaptation to only load the size we need)